lib/realtime/vehicle.ex

```diff
@@ -53,45 +68,52 @@ defmodule Realtime.Vehicle do
     :direction_id,
     :route_id,
     :trip_id,
+    :bearing,
+    :speed,
+    :stop_sequence,
+    :block_id,
+    :operator_id,
+    :operator_name,
+    :run_id,
     :stop_status,
     :timepoint_status
   ]
 
-  @doc """
-    Argument is an Elixir object. Pass it through Jason before this function.
-    json format for vehicles:
-    {
-      "id": "y0507",
-      "vehicle": {
-        "current_status": "IN_TRANSIT_TO",
-        "current_stop_sequence": 3,
-        "position": {
-          "bearing": 0,
-          "latitude": 42.35277354,
-          "longitude": -71.0593878
-        },
-        "stop_id": "6555",
-        "timestamp": 1554927574,
-        "trip": {
-          "direction_id": 0,
-          "route_id": "505",
-          "schedule_relationship": "SCHEDULED",
-          "start_date": "20190410",
-          "trip_id": "39984755"
+  @spec from_vehicle_position_and_trip_update(map() | nil, map() | nil) :: t()
+  def from_vehicle_position_and_trip_update(nil, _trip_update) do
+    nil
+  end
+
+  def from_vehicle_position_and_trip_update(_vehicle_position, nil) do
+    nil
+  end
+
+  def from_vehicle_position_and_trip_update(
+        %VehiclePosition{
```

Comment:
You shouldn't be matching the data out of the structs like this: they're generally treated as opaque. Instead, there are functions provided for you to get the relevant data: `VehiclePosition.id(vp)` &c.